pkg/clustermesh/endpointslicesync/clustermesh.go

```diff
@@ -124,26 +136,62 @@ func (cm *clusterMeshServiceGetter) GetServiceIP(svcID k8s.ServiceID) *loadbalan
 	return nil
 }
 
-func (cm *clusterMesh) newRemoteCluster(name string, status common.StatusFunc) common.RemoteCluster {
-	rc := &remoteCluster{
-		name:             name,
-		meshNodeInformer: cm.meshNodeInformer,
-		globalServices:   cm.globalServices,
-		storeFactory:     cm.storeFactory,
-		synced:           newSynced(),
-	}
+// RegisterClusterAddHook register a hook when a cluster is added to the mesh.
+// This should NOT be called after the Start hook.
```

Comment:
It may be a good idea to enforce this my panic'ing if called once the Start hook has been executed, to prevent silent failures in case of mistakes.